pkg/kubectl/resource_printer.go

```diff
@@ -452,8 +510,8 @@ var networkPolicyColumns = []string{"NAME", "POD-SELECTOR", "AGE"}
 
 // addDefaultHandlers adds print handlers for default Kubernetes types.
 func (h *HumanReadablePrinter) addDefaultHandlers() {
-	h.Handler(podColumns, printPod)
-	h.Handler(podColumns, printPodList)
+	h.Handler(podColumns, printPod, &h.hiddenPodsNum)
+	h.Handler(podColumns, printPodList, &h.hiddenPodsNum)
```

Comment:
Rather than passing in a pointer to `h.hiddenPodsNum`, try using a wrapper for the printFunc.
Could be either a private member of h or an anonymous function, something like
```
 func (h *HumanReadablePrinter) addDefaultHandlers() {
	h.Handler(podColumns, func(pod *api.Pod, w io.Writer, options PrintOptions) {
		reason := string(pod.Status.Phase)
		// if not printing all pods, skip terminated pods (default)
 		if !options.ShowAll && (reason == string(api.PodSucceeded) || reason == string(api.PodFailed)) {
			h.hiddenPods++
 			return nil
		}
		return printPod(pod, w, options)
	})
```

That way you don't have to change the interface or deal with reflection hacks.